src/fides/api/api/v1/endpoints/system.py

```diff
@@ -412,3 +426,16 @@ def instantiate_connection_from_template(
 
     system = get_system(db, fides_key)
     return instantiate_connection(db, saas_connector_type, template_values, system)
+
+
+async def supplement_privacy_declaration_response_with_legal_basis_override(resp: Union[Dict, System]) -> None:
+    """At runtime, adds a "legal_basis_for_processing_override" to each PrivacyDeclaration"""
+
+    for privacy_declaration in (
+        resp.get("privacy_declarations")
+        if isinstance(resp, Dict)
+        else resp.privacy_declarations
+    ):
+        privacy_declaration.legal_basis_for_processing_override = (
+            await privacy_declaration.overridden_legal_basis_for_processing
+        )
```

Comment:
I don't love this but am struggling to get async sessions to work with instance-level properties.   At any rate, the queries aren't efficient in these endpoints already and this doesn't help.  Because trying to return `privacy_declaration.overridden_legal_basis_for_processing` has errors with FastAPI code not awaiting the result, I am awaiting it here instead and stashing it on the PrivacyDeclaration under a different label.

Reply:
hmm. the major performance concerns should hopefully be mitigated pretty substantially by having a paged version of the endpoint that's relied upon for the UI (a separate ticket/unit of work), but yeah i see your qualms.

i do think that providing optionality (i.e. an endpoint query param) about whether this computed property is even returned at all could be another nice gate to put in place.

i'm also not 100% sure whether the `/consent/configure` page in the UI is still using our basic `/system` APIs - i know there had been some talk about making a specific endpoint for that page when we had some advanced filtering requirements. i think those requirements are no longer in place, but it's worth confirming that we're still planning to use this endpoint to back that UI. 

i'm getting a bit carried away in the details, but i think the major points are:
- this seems like a viable approach, at least as a starting point, whichever API this function ends up being used in
- let's keep an eye on the performance concerns, but they probably (hopefully?) will won't be too significant compared to other performance issues already in place on these endpoints, as you mention.
- if it adds nontrivial overhead, may be nice to provide optionality about whether the property is computed via an API param, since this is only needed for very particular purposes
- we should generally make sure we're all in sync on what endpoints are meant to back what APIs, there's been a lot of movement around all of this work over the past couple of weeks (cc @TheAndrewJackson @galvana)

Reply:
~also, a bit of a separate point - i could be missing something in the details, but do we have a mechanism for providing whether a purpose is broadly "excluded"? i.e. from what i can tell, here we are just computing the `overridden_legal_basis_for_processing`, but i don't think that provides a signal to the FE in the case where the purpose is just generally _excluded_ via an override?~

nevermind, i think i get it - if this is `None`/null, that indicates an exclusion, right?